src/controller.ts

```diff
@@ -215,6 +223,11 @@ class Controller {
       XMLHttpRequest,
     };
 
+    if (_endowments && _endowments.length > 0) {
+      _endowments.forEach((_endowment) => {
+        endowments[_endowment] = (window as any)[_endowment].bind(window);
+      });
```

Comment:
Some defensive improvements:
```suggestion
    if (_endowments && _endowments.length > 0) {
      _endowments.forEach((_endowment) => {
        if (!(_endowment in window)) {
          throw new Error(`Unknown endowment: "${_endowment}".`);
        }

        
        endowments[_endowment] = typeof globalValue === 'function'
          ? globalValue.bind(window)
          : globalValue;
      });
```